avr/cores/megacommand/WProgram.h

```diff
@@ -88,4 +88,25 @@ extern inline uint8_t switch_ram_bank(uint8_t x) {
   return x;
 }
 
+#ifdef __cplusplus
+
+class RamBankSelector {
+  private:
+  uint8_t m_oldbank;
+  public:
+  RamBankSelector(uint8_t bank) { m_oldbank = switch_ram_bank(bank); }
+  ~RamBankSelector() { switch_ram_bank(m_oldbank); }
+};
+
+#define select_bank(x) RamBankSelector __bank_selector(x)
+
+#endif// __cplusplus
+
+#define memcpy_bank1(dst, src, len) \
+  do { \
+  switch_ram_bank(1); \
+  memcpy(dst, src, len); \
+  switch_ram_bank(0); \
+  } while (false)
```

Comment:
what's the significance of "do { } while (false)" here? 🤔

Reply:
This is mainly a guard trick.
Without this, you could write:


if (something)
  memcpy_bank1(...)


when expanded, only the first statement will go into the conditional branch.

Reply:
I'm wondering what the performance overhead for destructor in ISR is.

You'll notice that I inline within the ISR routines wherever possible to avoid modifications to the current stack.

Reply:
most probably, the content of the destructor will be pulled out and inlined. this is a kind of zero-cost abstraction.